lib/_stream_readable.js

```diff
@@ -853,7 +853,15 @@ Readable.prototype.removeAllListeners = function(ev) {
 };
 
 function updateReadableListening(self) {
-  self._readableState.readableListening = self.listenerCount('readable') > 0;
+  const state = self._readableState;
+  state.readableListening = self.listenerCount('readable') > 0;
+  // try to start flowing for 'data' listeners
+  // (if pipesCount is not 0 then we have already 'flowed')
+  if (!state.readableListening &&
+      self.listenerCount('data') > 0 &&
+      state.pipesCount === 0) {
+    self.resume();
```

Comment:
what if the stream was explicitly paused before? Something like:

``` js
stream.pause()
stream.on('data', ...)
stream.once('readable', ...)
```

Seems bad that this would auto resume that

Reply:
@lundibundi can you please add a unit test in that case?

Reply:
> I'm thinking of not setting flowing at all until we have 0 'readable' listeners at which point check for data listeners and then set it to true/false (resume/pause).

I think that would work.